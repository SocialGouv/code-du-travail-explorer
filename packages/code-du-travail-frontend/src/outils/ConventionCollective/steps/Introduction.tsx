import {
  AlertWithIcon,
  Button,
  Text,
  theme,
  Tile as TileUi,
} from "@socialgouv/cdtn-ui";
import React from "react";
import styled from "styled-components";

import {} from "../../common/type/WizardType";
import { SearchType } from "..";

type Props = {
  onSelecSearchType: (type: SearchType) => void;
};

const IntroductionStep = ({ onSelecSearchType }: Props): JSX.Element => {
  return (
    <>
      <AlertWithIcon variant="secondary">
        <p>
          La convention collective est un texte conclu au niveau d’une branche
          d’activité (Ex : Transports routiers). Elle adapte les règles du code
          du travail aux situations particulières de la branche (primes, congés,
          salaires minima, préavis, prévoyance…).
        </p>
        <Text as="p" fontWeight="700">
          Vous pouvez retrouver le nom de votre convention collective sur votre
          bulletin de paie ou sur votre contrat de travail.
        </Text>
      </AlertWithIcon>
      <Flex>
        <Tile onClick={() => onSelecSearchType(SearchType.agreement)}>
          <Text fontWeight="700">
            Je connais
            <br /> ma convention collective
          </Text>
          <ButtonWrapper>
            <Button variant="link" as="div">
              Je la saisie
            </Button>
          </ButtonWrapper>
        </Tile>
        <Tile onClick={() => onSelecSearchType(SearchType.compagny)}>
          <Text fontWeight="700">
            Je ne connais <br /> pas ma convention collective
          </Text>
          <ButtonWrapper>
            <Button variant="link" as="div">
              Je la recherche
            </Button>
          </ButtonWrapper>
        </Tile>
      </Flex>
    </>
  );
};

export { IntroductionStep };

const Flex = styled.div`
  display: flex;
  padding-top: ${theme.spacings.medium};
  justify-content: space-around;
  @media (max-width: ${theme.breakpoints.mobile}) {
    flex-direction: column;
    align-items: center;
  }
`;

const Tile = styled(TileUi)`
  flex: 0 1 auto;
  width: 28rem;
  @media (max-width: ${theme.breakpoints.mobile}) {
    & + & {
      margin-top: ${theme.spacings.medium};
    }
  }
`;

const ButtonWrapper = styled.div`
  padding-top: ${theme.spacings.medium};
`;
