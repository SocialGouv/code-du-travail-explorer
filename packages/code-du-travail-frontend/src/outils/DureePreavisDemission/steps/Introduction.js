import React from "react";

function StepIntro() {
  return (
    <>
      <p>
        En cas de démission, le salarié ne peut quitter l’entreprise
        immédiatement après avoir informé l’employeur de ses intentions. Il doit
        rester dans l’entreprise durant une certaine période, qu’on appelle
        préavis ou parfois délai congé.
      </p>

      <p>
        Les règles encadrant le préavis (droits et obligations, durées, cas de
        dispense, absences autorisées pour chercher un emploi durant le préavis)
        sont notamment définies par le code du travail et la branche.
      </p>

      <p>
        Le présent outil vous permet de connaitre la durée du préavis prévue par
        votre convention collective en matière de démission. Prévoyez une à deux
        minutes pour cette simulation.
      </p>
    </>
  );
}

export { StepIntro };
