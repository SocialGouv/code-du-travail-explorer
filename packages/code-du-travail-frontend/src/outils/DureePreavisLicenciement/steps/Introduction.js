import React from "react";
import Link from "next/link";

function StepIntro() {
  return (
    <>
      <p>
        Le préavis est une période accordée au salarié entre la notification du
        licenciement et son départ effectif de l’entreprise (fin du contrat).
      </p>

      <p>
        Afin de réaliser cette simulation, vous aurez besoin d'informations
        concernant la situation de l'employé concerné, comme sa convention
        collective, sa catégorie ou son coefficient. La plupart de ces
        informations se trouvent sur le bulletin de salaire ou le contrat de
        travail. Prévoyez deux à cinq minutes pour cette simulation.
      </p>

      <p>
        Vous pouvez trouver plus d'informations sur le préavis de licenciement
        sur{" "}
        <Link
          href="/fiche-service-public/[slug]"
          as={`/fiche-service-public/preavis-de-licenciement`}
        >
          <a>cette fiche</a>
        </Link>
        .
      </p>
      <p>Cliquez sur suivant pour commencer la simulation</p>
    </>
  );
}

export { StepIntro };
