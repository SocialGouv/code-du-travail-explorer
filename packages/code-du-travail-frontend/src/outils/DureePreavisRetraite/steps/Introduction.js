import React from "react";

function StepIntro() {
  return (
    <>
      <p>
        Le préavis est une période accordée au salarié entre la notification du
        départ ou la mise à retraite et son départ effectif de l’entreprise (fin
        du contrat).
      </p>

      <p>
        Afin de réaliser cette simulation, vous aurez besoin d’informations
        concernant la situation de l’employé concerné, comme sa convention
        collective, sa catégorie ou son coefficient. La plupart de ces
        informations se trouvent sur le bulletin de salaire ou le contrat de
        travail. Prévoyez 2 à 5 minutes pour cette simulation.
      </p>

      <p>Cliquez sur commencer pour démarrer la simulation.</p>
    </>
  );
}

export { StepIntro };
