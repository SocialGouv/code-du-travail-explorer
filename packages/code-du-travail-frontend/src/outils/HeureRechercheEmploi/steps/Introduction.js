import React from "react";

function StepIntro() {
  return (
    <>
      <p>
        Ce simulateur vous permet de savoir si votre convention collective a
        prévu la possibilité pour le salarié de s’absenter en fin de contrat,
        durant le préavis, pour rechercher un emploi.
      </p>

      <p>
        Le code du travail ne prévoit pas ce droit ( sauf cas particulier). En
        revanche, une majorité de conventions collective fixent pour ces heures
        d’absences: leur nombres et modalités de mise en oeuvre. Autant
        d’informations que vous trouverez sur cet outil.
      </p>

      <p>
        Afin de réaliser cette simulation, vous aurez besoin d’informations
        concernant la situation du salarié concerné, comme sa convention
        collective, sa catégorie, son ancienneté. La plupart de ces informations
        se trouvent sur le contrat de travail ou le bulletin de salaire.
        Prévoyez deux à cinq minutes pour cette simulation.
      </p>
      <p>Cliquez sur suivant pour commencer la simulation</p>
    </>
  );
}

export { StepIntro };
