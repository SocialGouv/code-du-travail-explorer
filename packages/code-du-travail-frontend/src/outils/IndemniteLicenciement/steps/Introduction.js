import React from "react";
import Link from "next/link";

const StepIntro = () => (
  <>
    <p>
      Ce simulateur est un outil qui permet d’estimer le montant de l’indemnité
      minimale de licenciement d’un salarié.
    </p>
    {/* <p>
      Vous pourrez simuler simplement le montant de l’indemnité légale définie
      par le Code du travail et le montant de l’indemnité conventionnelle de
      licenciement si la branche d’activité prévoit un montant supérieur au Code
      du travail.
    </p> */}
    <p>
      Cette simulation nécessite entre 5 et 10 minutes. Afin de pouvoir remplir
      les renseignements demandés, munissez-vous des informations relatives au
      licenciement (dates d’entrée et de sortie de l’entreprise, courrier de
      licenciement, derniers bulletins de salaire, etc.).
    </p>
    <p>
      Pour en savoir plus sur l’indemnité de licenciement et son mode de calcul,
      consultez{" "}
      <Link
        href="/fiche-service-public/[slug]"
        as={`/fiche-service-public/indemnite-de-licenciement`}
      >
        <a>cet article</a>
      </Link>
      .
    </p>
    <p>Pour commencer la simulation cliquez sur &quot;Suivant&quot;.</p>
  </>
);

export { StepIntro };
