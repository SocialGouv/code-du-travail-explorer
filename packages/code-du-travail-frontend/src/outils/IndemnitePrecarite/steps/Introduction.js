import React from "react";

function StepIntro() {
  return (
    <>
      <p>
        Ce simulateur est un outil qui permet d’estimer le montant de
        l’indemnité de fin de contrat ou de fin de mission (dite “prime de
        précarité”) d’un salarié en CDD ou en contrat d’intérim.
      </p>
      <p>
        Vous pourrez simuler simplement le montant de la prime de précarité
        définie par le Code du travail avec la prise en compte des
        particularités présentes dans certaines branches d’activité.
      </p>
      <p>
        Cette simulation nécessite environ 5 minutes. Afin de pouvoir remplir
        les renseignements demandés, munissez-vous des informations relatives à
        votre contrat de travail (CDD ou contrat de travail temporaire (contrat
        d’intérim), derniers bulletins de salaire, …).
      </p>
      <p>Pour commencer la simulation, cliquez sur « Suivant ».</p>
    </>
  );
}

export { StepIntro };
