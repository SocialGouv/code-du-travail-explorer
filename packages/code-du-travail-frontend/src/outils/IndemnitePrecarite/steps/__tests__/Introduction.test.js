import React from "react";
import { render } from "@testing-library/react";
import { StepIntro } from "../Introduction";

describe("<StepIntro />", () => {
  it("should render", () => {
    const { container } = render(<StepIntro />);
    expect(container).toMatchSnapshot();
  });
});
