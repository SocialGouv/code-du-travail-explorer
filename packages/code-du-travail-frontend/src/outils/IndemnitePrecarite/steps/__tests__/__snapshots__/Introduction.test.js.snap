// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StepIntro /> should render 1`] = `
<div>
  <p>
    Ce simulateur est un outil qui permet d’estimer le montant de l’indemnité de fin de contrat ou de fin de mission (dite “prime de précarité”) d’un salarié en CDD ou en contrat d’intérim.
  </p>
  <p>
    Vous pourrez simuler simplement le montant de la prime de précarité définie par le Code du travail avec la prise en compte des particularités présentes dans certaines branches d’activité.
  </p>
  <p>
    Cette simulation nécessite environ 5 minutes. Afin de pouvoir remplir les renseignements demandés, munissez-vous des informations relatives au contrat de travail (CDD ou contrat de travail temporaire (contrat d’intérim), derniers bulletins de salaire, …).
  </p>
  <p>
    Pour commencer la simulation, cliquez sur « Suivant ».
  </p>
</div>
`;
